jwst/spectral_leak/spectral_leak_step.py

```diff
@@ -0,0 +1,94 @@
+#! /usr/bin/env python
+
+from stdatamodels.jwst import datamodels
+from jwst.datamodels import ModelContainer
+from ..stpipe import Step
+import numpy as np
+
+from . import spectral_leak
+
+__all__ = ["SpectralLeakStep"]
+
+
+class SpectralLeakStep(Step):
+    """
+    The MIRI MRS has a spectral leak in which 6 micron light leaks into the
+    12 micron channel.  This step applies a correction to the 12 micron channel.
+    """
+
+    class_alias = "spectral_leak"
+
+    reference_file_types = ['mrsptcorr']
+
+    def process(self, input):
+        """Execute the step.
+
+        Parameters
+        ----------
+        input : container of models containing 1-D extracted spectra
+
+        Returns
+        -------
+        JWST DataModel
+            This will be `input` if the step was skipped; otherwise,
+            it will be a corrected 1-D extracted spectra that contains
+            the  3B MRS range.
+        """
+
+        ch1b = None
+        ch3a = None
+        ich3a = None
+        ch1b_wave = 6.0
+        ch3a_wave = 12.0
+
+        result = input.copy()  # copy input to return
+        with datamodels.open(input) as input_model:
+            if isinstance(input_model, ModelContainer):
+                # Retrieve the reference parameters for this type of data
+                sp_leak_ref = self.get_reference_file(input[0], 'mrsptcorr')
+
+                for i, x1d in enumerate(input_model):  # input_models is a Model Container
+                    # check that we have the correct type of data
+                    if not isinstance(x1d, datamodels.MultiSpecModel):
+                        self.log.warning("Data sent to spectral_leak step is not an extracted spectrum. It is  {}." .format(type(x1d)))
+                        return
+
+                    channel = x1d.meta.instrument.channel
+                    band = x1d.meta.instrument.band
+                    srctype = x1d.spec[0].source_type
+                    if srctype == 'EXTENDED':
+                        self.log.info('No spectral leak correction for extended source data')
+                        return input
+                    # search x1d containing CH 1 B
+                    if '1' in channel and 'MEDIUM' in band:
+                        print('found ch 1B')
```

Comment:
Should this and other print statements below be removed, or use the usual logging stream?

Reply:
Whoops I forgot to remove that print statement. I will do that.